Sources/Client/Client/Client+WebSocket.swift

```diff
@@ -53,13 +53,33 @@ extension Client {
             
             if case .connected(let userConnection) = connectionState {
                 self.userAtomic.set(userConnection.user)
+                self.restoreWatchingChannels()
                 
                 if self.isExpiredTokenInProgress {
                     self.performInCallbackQueue { [unowned self] in self.sendWaitingRequests() }
                 }
             }
         }
     }
+    
+    private func restoreWatchingChannels() {
+        watchingChannelsAtomic.flush()
+        
+        guard let keys = watchingChannelsAtomic.get()?.keys, !keys.isEmpty else {
+            return
+        }
+        
+        let cids = Array(keys).chunked(into: 50)
+        
+        cids.forEach { chunk in
+            eventsHandlingQueue.asyncAfter(deadline: .now() + .milliseconds(Int.random(in: 0...500))) { [unowned self] in
```

Comment:
why not keep this simple and do a simple for-loop?

Reply:
Fixed